doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,467 @@
+        * information has been collected (e.g. the report metadata).
+        * @param uri URI to compartment report. The report type is deduced from
+        *        here.
+        * @param mode the brion::AccessMode bitmask
+        * @param gids the neurons of interest in READ_MODE
+        * @throw std::runtime_error if source is not a valid compartment report
+        *        or could not be open in the mode requested.
+        * @version TBD
+        */
+       CompartmentReport( const URI& uri, const int mode )
+
+       /*
+        * Open a report in read mode with a subset selection.
+        *
+        * This function may block if needed until all the information has
+        * been collected (e.g. the report metadata).
+        *
+        * @param uri URI to compartment report. The report type is deduced from
+        *        here.
+        * @param subset The set of gids to be reported. This set should be
+        *        understood as a filter.
+        * @throw std::runtime_error if source is not a valid compartment report
+        *        or could not be open.
+        * @version TBD
+        */
+       CompartmentReport( const URI& uri, const GIDSet& subset );
+
+  These constructors are also consistent with the new SpikeReport.
+
+- Add the following functions:
+
+       /**
+        * @return The current position in the report. This is the start time
+        *         to use in write operations and the next read operation.
+        * @version TBD
+        */
+       float getCurrentTime() const;
+
+       enum State { OK = 0, ENDED = 1, FAILED = 2 };
+       /**
+        * @return The state after the last completed operation.
+        * @version TBD
+        */
+       State getState() const;
+
+       /**
+        * Forward to a given absolute timestamp.
+        *
+        * Preconditions:
+        * - r.getState() is OK or ENDED.
+        * - The report was open in read mode.
+        * - There is no previous read or seek operation with a pending future.
+        * - toTimestamp >= getStartTime()
+        *
+        * Postconditions:
+        * Let:
+        *  - r be the CompartmentReport
+        *  - f be the returned future by r.seek(toTimestamp)
+        *  After f.wait() returns:
+        *  - r.getCurrentTime() ~= toTimestamp rounded to the nearest multiple
+             of the timestep, ...
+        *  - ... getState() == OK -> r.getEndTime() > getCurrentTime() and ...
+        *  - ... getState() == ENDED -> r.getEndTime() == getCurrentTime()
+        *
+        * After successful f.wait_for or f.wait_until, the result is the same
+        * as above. If they time out, no observable state changes.
+        *
+        * @throw std::logic_error if a precondition is not met..
+        * @version TBD
+        */
+       boost::future< void > seek( const float toTimestamp );
+
+       /** Finish writing the current frame and advance to the next one.
```

Comment:
The class description has been extended. Let me know if that's sufficient.
Note that the API functions that refer to frames do not use frame numbers at all. This means that how timestamps are discretized into frames is kept inside the implementation.